### PR TITLE
Send Node Architecture with AKS Init Data

### PR DESCRIPTION
**Pull request by fals**
- Added new field to request
- Added unit test

### CODE REVIEW

File: castai/types.go [hunk 1]
```diff
@@ -186,7 +186,15 @@ func (c *ChartSource) Validate() error {
 	return nil
 }
 
+type NodeArchitecture int32
+
+const (
+	NodeArchitectureAmd64 NodeArchitecture = 0
+	NodeArchitectureArm64 NodeArchitecture = 1
```
Comment (varnastadeus) [inline castai/types.go:193]: why not just sending `arm64/amd64` instead? 🤔

Comment (fals) [reply castai/types.go:193]: I designed the api to not rely on strings, I think it is easier to transform and make operations.

Comment (varnastadeus) [reply castai/types.go:193]: But it is much easier to reason about as 0/1 doesn't says much and you need to do additional mapping everywhere down the pipeline

Comment (fals) [reply castai/types.go]: changed as suggested

File: castai/types.go [hunk 2]
```diff
@@ -187,6 +187,7 @@ func (c *ChartSource) Validate() error {
 }
 
 type AKSInitDataRequest struct {
-	CloudConfigBase64       string `json:"cloudConfigBase64"`
-	ProtectedSettingsBase64 string `json:"protectedSettingsBase64"`
+	CloudConfigBase64       string  `json:"cloudConfigBase64"`
+	ProtectedSettingsBase64 string  `json:"protectedSettingsBase64"`
+	Architecture            *string `json:"architecture,omitempty"`
```
Comment (varnastadeus) [inline castai/types.go:192]: why its nullable?

Comment (fals) [reply castai/types.go:192]: IMO Everything new should be optional. What is the concern?

Comment (fals) [reply castai/types.go]: Just added as suggested

**Review by varnastadeus — APPROVED** [review]

**Pull request merged**